TheRealJournalRando/Resources/Logic/waypoints.json

```diff
@@ -338,7 +338,7 @@
   },
   {
     "name": "Defeated_Any_Belfly",
-    "logic": "*Pale_Ore-Basin | Deepnest_East_07 | Deepnest_East_08[right1] | Deepnest_East_09[left1] | RestingGrounds_10 + UPWALLBREAK | Room_Colosseum_01[left1] + (Defeated_Colosseum_2 | Defeated_Colosseum_3) | Ruins1_01[left1] | Ruins1_05[bot1] + (LEFTCLAW | WINGS + (RIGHTCLAW | ENEMYPOGOS)) | Ruins1_05c | Ruins2_07[right1] | Waterways_01 | Waterways_02[top1] | Waterways_02[top2] | (Waterways_02[bot2] | Waterways_02[top3] + Broke_Waterways_Bench_Quake_Floor_3) + (ANYCLAW | WINGS | ENEMYPOGOS) | Waterways_02[bot1] + (ANYCLAW | WINGS) | Waterways_04[left2] + (SWIM | RIGHTSKIPACID) | Waterways_04[right1] | Waterways_04[left1] + (RIGHTCLAW | WINGS | ENEMYPOGOS) | *Rancid_Egg-Waterways_West_Pickup | Waterways_07 | *Soul_Totem-Waterways_Flukemarm | Waterways_08[left2] | Waterways_13[left1]",
+    "logic": "*Pale_Ore-Basin | Deepnest_East_07 | Deepnest_East_08[right1] | Deepnest_East_09[left1] | RestingGrounds_10 + (Wall-Catacombs_Left_1?UPWALLBREAK | Wall-Catacombs_Right_1?FALSE + Wall-Catacombs_Grub?FALSE) | Room_Colosseum_01[left1] + (Defeated_Colosseum_2 | Defeated_Colosseum_3) | Ruins1_01[left1] | Ruins1_05[bot1] + (LEFTCLAW | WINGS + (RIGHTCLAW | ENEMYPOGOS)) | Ruins1_05c | Ruins2_07[right1] | Waterways_01 | Waterways_02[top1] | Waterways_02[top2] | (Waterways_02[bot2] | Waterways_02[top3] + Broke_Waterways_Bench_Quake_Floor_3) + (ANYCLAW | WINGS | ENEMYPOGOS) | Waterways_02[bot1] + (ANYCLAW | WINGS) | Waterways_04[left2] + (SWIM | RIGHTSKIPACID) | Waterways_04[right1] | Waterways_04[left1] + (RIGHTCLAW | WINGS | ENEMYPOGOS) | *Rancid_Egg-Waterways_West_Pickup | Waterways_07 | *Soul_Totem-Waterways_Flukemarm | Waterways_08[left2] | Waterways_13[left1]",
```

Comment:
For the right side belfly by the grub, shouldn't the falses be "can break the walls" e.g. upwallbreak or similar? Additionally does that belfly need vertical to trigger?

Reply:
It could be UPWALLBREAK as an alternative to FALSE. In truth, either all of the wall terms exist or none does, so if they aren't present, the coalesce is (UPWALLBREAK | FALSE), which is kind of the same as UPWALLBREAK | UPWALLBREAK. It can be changed but should have no impact.

No movement is required for those belflys to trigger.

Reply:
I think using UPWALLBREAK should be preferable as walls could conceivably be custom pool injected so the assumption they all exist together may not hold up

Reply:
So I guess it should be right_1?true + grub? upwallbreak

Reply:
Further reviewing this, I reworked the general way BWR handles the RestingGrounds_10 waypoint, as the previous approach led to a lot of false negatives and could get to false positives if not properly handled. Since Belflys can be fought on both sides of the room, the Catacombs_Right_1 wall shouldn't be mandatory, so I removed it from the check entirely and it should be included in the "RestingGrounds_10" token. Using the transitions instead of the full waypoint should prevent all false positive and negative cases.